apps/nestjs-api/src/auth-entra-id/entra-id.service.ts

```diff
@@ -0,0 +1,149 @@
+import {
+  AuthenticationResult,
+  AuthorizationCodeRequest,
+  ClientApplication,
+  ConfidentialClientApplication,
+  Configuration,
+  CryptoProvider,
+} from '@azure/msal-node'
+import { HttpService } from '@nestjs/axios'
+import { Injectable } from '@nestjs/common'
+import { AxiosError, AxiosRequestConfig } from 'axios'
+import { NextFunction, Request, Response } from 'express'
+import { firstValueFrom } from 'rxjs'
+import { catchError } from 'rxjs/operators'
+import { EntraIdConfigProvider } from './entra-id-config.provider'
+import { VerificationData } from './verification-data.interface'
+
+@Injectable()
+export class EntraIdService {
+  readonly verifierCookieName = 'entra_id_verifier'
+  private readonly cryptoProvider = new CryptoProvider()
+
+  constructor(
+    private readonly configProvider: EntraIdConfigProvider,
+    private readonly httpService: HttpService
+  ) {}
+
+  async getClientApplication(): Promise<ClientApplication> {
+    const config = await this.prepareConfig()
+
+    return new Promise((resolve) => {
```

Comment:
Isn't this just boilerplate for the simpler ...
```
return new ConfidentialClientApplication(config)
```
?

Reply:
😆 It totally is. I think I was desperately trying to fix something and trying a few crazy things and then forgot to take this one out. 😆